api/v1alpha1/hostedcluster_types.go

```diff
@@ -637,6 +637,7 @@ type AzurePlatformSpec struct {
 	ResourceGroupName string                      `json:"resourceGroup"`
 	VnetName          string                      `json:"vnetName"`
 	VnetID            string                      `json:"vnetID"`
+	SubnetName        string                      `json:"subnetName"`
```

Comment:
why is this here instead of in NodePool API?

Reply:
Because the assumption so far is that there is one subnet per cluster

Reply:
Is that the case with standalone OCP?

Reply:
Standalone OCP creates two subnets, one for the masters, one for the workers: https://github.com/openshift/installer/blob/8fca1ade5b096d9b2cd312c4599881d099439288/data/data/azure/vnet/vnet.tf

But since we don't have masters, we don't need the former.

The [main reasons to create multiple subnets seem to be](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-vnet-plan-design-arm#subnets):
a) Some azure resources create their own subnet
b) Security, network security groups are associated to a subnet

But since from an azure networking perspective all workers fall into the same security domain (they need to be able to talk to each other freely, reach the internet and be reachable by LBs) I don't see why we would need multiple. Cluster-level network security happens below azure through the clusters SDN.

Reply:
We should possibly expose the [AZ within the subnet](https://docs.microsoft.com/en-us/azure/availability-zones/az-overview#availability-zones) https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azuremachine_types.go#L44.
This is also what we do in the [installer](https://github.com/openshift/installer/blob/master/pkg/asset/machines/worker.go#L360).

Reply:
What do you mean by "AZ within the subnet"? The subnet is not scoped to an AZ, it is possible to put an instance in a different AZ but use the same subnet. There is a follow-up task for Azure multi-zone support: https://issues.redhat.com/browse/HOSTEDCP-322

Reply:
yeh I meant the list of az in that subnet, cool yeah so basically 332